### DISCUSSION TITLE
How to use UpdateStatus

### DISCUSSION THREAD

**Discussion by basilfx** [buehler/dotnet-kubernetes-client#37] [Q&A]
I'm trying to update the status of a (custom) entity, but I don't see how I should use it properly. Can you provide an example?

I have the following code:

```csharp
var deployment = new V1Deployment()
{
    ...
}

await _kubernetesClient.UpdateStatus(deployment);
```

Which gives me the following error:

```
The type 'k8s.Models.V1Deployment' cannot be used as type parameter 'TResource' in the generic type or method 'IKubernetesClient.UpdateStatus<TResource>(TResource)'. There is no implicit reference conversion from 'k8s.Models.V1Deployment' to 'k8s.IStatus<object>'.
```

I have the same problem with a custom entity that was defined as follows:

```csharp
public class V1MyEntity : CustomKubernetesEntity<V1Alpha1ConnectorSetSpec, V1Alpha1ConnectorSetStatus>
{
    ...
}
```

I think this error is because `UpdateStatus` expects an instance that implements `IStatus<object>`: https://github.com/buehler/dotnet-kubernetes-client/blob/master/src/DotnetKubernetesClient/IKubernetesClient.cs#L130

I'm using this library via the `dotnet-operator-sdk` (version 4.0.20).

**Comment by basilfx** [comment]
@buehler can you help me with this question?

**Reply by buehler** [reply]
Hey @basilfx 

Sorry, I did not see the question in the first place!

Exactly, to update the status of an object, it must inherit the status interface. At least this is what I was expecting. The reason is, that CRD's are generated in a unique way with a status object. If this object is not generated with the updateable status object, updating the status only is not possible.

But when you use the custom entity, it should generally be possible (when you generate the CRD with my SDK at least)

Do you use the .net sdk (https://github.com/buehler/dotnet-operator-sdk)?

**Reply by buehler** [reply]
When we have a look at the code:
https://github.com/buehler/dotnet-kubernetes-client/blob/master/src/DotnetKubernetesClient/KubernetesClient.cs#L245-L258

This uses "ReplaceClusterCustomObjectStatus"

I just had a look at the base code... in theory, we could insert an object there (instead of `IStatus`).
Maybe this would work.

**Reply by basilfx** [reply]
I don't see how that works.

I am using dotnet-operator-sdk, and the custom entity I am using is as shown above. Do you have an example that works?

**Reply by buehler** [reply]
Hmm weird. yes: https://github.com/smartive/hosted-database-operator/blob/f75a127fc32dddd7268cc5ad830c387a274ddd80/src/HostedDatabaseOperator/Controller/ClusterDatabaseHostController.cs#L94-L127 this is working. But I have a CRD instead of V1Deployment.

I think I need to investigate this matter
